### PR TITLE
Add deployed status to sites

### PR DESCRIPTION
**Pull request by ericboucher**
- Add deployed status flag to sites
- When filtering by spotters only, only display sites marked as deployed
- On the site details page, do not load sensor data for non-admins if the buoy is not deployed, we achieve this by only sending spotter data through the API if the buoy is deployed.

### CODE REVIEW

File: packages/api/src/utils/liveData.ts [hunk 1]
```diff
@@ -87,21 +87,22 @@ export const getLiveData = async (
...
-        wavePeakPeriod: getLatestData(spotterRawData.wavePeakPeriod),
-        waveMeanDirection: getLatestData(spotterRawData.waveMeanDirection),
-        longitude:
-          spotterRawData.longitude && getLatestData(spotterRawData.longitude),
-        latitude:
-          spotterRawData.latitude && getLatestData(spotterRawData.latitude),
-      }
-    : {};
+  const spotterData =
+    spotterRawData && reef.status === 'deployed'
```
Comment (Ry-DS) [inline packages/api/src/utils/liveData.ts:91]: Shouldn't we use the enum here to compare? Like `reef.status === ReefStatus.Deployed`

File: packages/website/src/routes/ReefRoutes/Reef/Sensor/index.tsx [hunk 2]
```diff
@@ -47,6 +47,10 @@ const applicationTag = (
```
Comment (Ry-DS) [inline packages/website/src/routes/ReefRoutes/Reef/Sensor/index.tsx:47]: While we are here, maybe its worthwhile to give this function a better name like `getApplicationTag` or `findApplicationTag` along with a docs comment? Not sure what the last boolean here represents at a glance.

**Review by Ry-DS — APPROVED** [review]
Overall looks good! Just a few small comments to improve readability

File: packages/api/src/utils/liveData.ts [hunk 3]
```diff
@@ -87,21 +87,22 @@ export const getLiveData = async (
...
+      ? {
+          surfaceTemperature: getLatestData(spotterRawData.surfaceTemperature),
```
Comment (Ry-DS) [inline packages/api/src/utils/liveData.ts:93]: I would personally use lodash's `mapValues` util function; it looks like we are just taking every field from `spotterRawData` and running it through `getLatestData`. What do you think?

Comment (ericboucher) [reply packages/api/src/utils/liveData.ts:93]: You're thinking of something like this?

```
const spotterData =
    spotterRawData && reef.status === ReefStatus.Deployed
      ? mapValues(spotterRawData, (values) => values && getLatestData(values))
      : {};
```

Comment (Ry-DS) [reply packages/api/src/utils/liveData.ts]: This works for me: 
```ts
const spotterData: { [k in keyof SpotterData]?: SofarValue } =
    spotterRawData && reef.status === ReefStatus.Deployed
      ? mapValues(spotterRawData, (values) => values && getLatestData(values))
      : {};
```

File: packages/api/src/reefs/reefs.service.ts [hunk 4]
```diff
@@ -210,24 +210,26 @@ export class ReefsService {
...
-      );
+        reef.status === ReefStatus.Deployed &&
+        isNil(
+          await this.exclusionDatesRepository
+            .createQueryBuilder('exclusion')
+            .where('exclusion.spotter_id = :spotterId', {
+              spotterId: reef.spotterId,
+            })
+            .andWhere('DATE(exclusion.startDate) <= DATE(:date)', {
+              date: new Date(),
```
Comment (Ry-DS) [inline packages/api/src/reefs/reefs.service.ts:223]: `new Date()` is used 3 times, might be worthwhile to move it to `const now=new Date()`?

Comment (ericboucher) [reply packages/api/src/reefs/reefs.service.ts:223]: I think these functions will disappear very soon (we're changing how we handle exclusion dates)

Comment (Ry-DS) [reply packages/api/src/reefs/reefs.service.ts:223]: Sounds good :+1:

**Review by tleraj — APPROVED** [review]
Everything looks good!!

**Comment by tleraj** [comment]
Noticed a text break on the sensor card on`960px` width.
Screenshot_20210118_160600

**Pull request merged**